js/contextMenus.js

```diff
@@ -374,6 +374,35 @@ function usernameTemplateInit (usernames, origin, action) {
   return items
 }
 
+function autofillTemplateInit (suggestions, frame) {
+  let items = []
+  for (let i = 0; i < suggestions.length; ++i) {
+    let value
+    let frontendId = suggestions[i].frontend_id
+    if (frontendId >= 0) { //  POPUP_ITEM_ID_AUTOCOMPLETE_ENTRY and Autofill Entry
+      value = suggestions[i].value
+    } else if (frontendId === -1) { // POPUP_ITEM_ID_WARNING_MESSAGE
+      value = 'Disabled due to unsecure connection.'
+    } else if (frontendId === -4) { // POPUP_ITEM_ID_CLEAR_FORM
+      value = 'Clear Form'
+    } else if (frontendId === -5) { // POPUP_ITEM_ID_AUTOFILL_OPTIONS
+      value = 'Autofill Settings'
+    }
+    if (frontendId === -3) { // POPUP_ITEM_ID_SEPARATOR
+      items.push(CommonMenu.separatorMenuItem)
+    } else {
+      items.push({
+        label: value,
+        click: (item, focusedWindow) => {
+          ipc.send('autofill-selection-clicked', frame.get('tabId'), value, frontendId, i)
```

Comment:
can we get rid of the ipc message and do a window action here that calls `webContents.autofillSelect` in frames.js? Maybe track the autofill status in the frame state and show/hide the context menu based on that? Could definitely be done in a follow-up commit, but we will have to refactor the ipc eventually anyway when moving webContents mgmt to the browser process